src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTableUI.java

```diff
@@ -843,4 +852,37 @@ public void paint(Graphics g) {
             SynthLookAndFeel.resetSelectedUI();
         }
     }
+    private int getXPosition(int column) {
```

Comment:
This method is exactly the same as in `BasicTableUI`, if I don't miss anything.

You should rather move this to a utility class which is accessible to both `-UI` classes. Even more: if you're following the same code path that's implemented for `JTableHeader`, you should consider re-using the code from the header painting too… by extracting the relevant parts to a utility class if appropriate.

If another bug is found, it will need to be fixed in *one place* instead of several places which repeat the same code.

Reply:
The methods can be both moved to Utility class for sure, but not sure of re-using the code from `JTableHeader` coz `getColumnModel()` and `getWidthInRightToLeft()` are specific to it.

Reply:
How is `getColumnModel()` specific?

I didn't dive into comparing rendering… I expect the gist of it is the same, you mentioned that it was similar. It doesn't necessarily mean the header and table cells can re-use the algorithm, yet I feel it will be beneficial — one algorithm for the two similar cases, less code duplication.